java/vector/src/test/java/org/apache/arrow/vector/ipc/TestArrowFile.java

```diff
@@ -708,6 +711,56 @@ public void testWriteReadVarBin() throws IOException {
     }
   }
 
+  @Test
+  public void testReadWriteMultipleBatches() throws IOException {
+    File file = new File("target/mytest_nulls_multibatch.arrow");
+
+    try (IntVector vector = new IntVector("foo", allocator);) {
+      Schema schema = new Schema(Collections.singletonList(vector.getField()), null);
+      try (FileOutputStream fileOutputStream = new FileOutputStream(file);
+           VectorSchemaRoot root = new VectorSchemaRoot(schema, Collections.singletonList((FieldVector) vector), vector.getValueCount());
+           ArrowFileWriter writer = new ArrowFileWriter(root, new MapDictionaryProvider(), fileOutputStream.getChannel());) {
+        writer.start();
+
+        vector.setNull(0);
+        vector.setSafe(1, 1);
+        vector.setSafe(2, 2);
+        vector.setNull(3);
+        vector.setSafe(4, 1);
+        vector.setValueCount(5);
+        root.setRowCount(5);
+        writer.writeBatch();
+
+        vector.setNull(0);
+        vector.setSafe(1, 1);
+        vector.setSafe(2, 2);
+        vector.setValueCount(3);
+        root.setRowCount(3);
+        writer.writeBatch();
```

Comment:
You should be able to put the body of this in BaseFileTest and then use the same thing for ArrowFileReader and ArrowStreamReader